### PR TITLE
TCM-272 | Build dummy videos screen

### PR DESCRIPTION
**Pull request by halilyesilyurt**
### Summary

- [x] Completed dummy videos screen.
- [x] Added videos screen navigation to video screen.

### Issue

[TCM-272](https://hyfnla.atlassian.net/browse/TCM-272)

### Video/Screenshot

| BEFORE_SCROLL| AFTER_SCROLL|
|--------|--------|
|videos_1|videos_2|

DEMO

https://user-images.githubusercontent.com/30659854/118697665-3736db80-b818-11eb-8fda-8e10d726cf2b.mp4

### CODE REVIEW

**Comment by halilyesilyurt** [comment]
updated pr images and added demo.

File: app/src/main/java/com/trx/consumer/models/params/VideoParamsModel.kt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.trx.consumer.models.params
+
+import android.os.Parcelable
+import com.trx.consumer.models.common.WorkoutModel
+import kotlinx.parcelize.Parcelize
+
+@Parcelize
+data class VideoParamsModel(
```
Comment (VivekV95) [inline app/src/main/java/com/trx/consumer/models/params/VideoParamsModel.kt:8]: Don't think `VideosFragment` will be receiving a `WorkoutModel`, let's create a `VideosModel` that will have the following properties:
```
@Parcelize
class VideosModel(
    val title: String = "",
    val poster: String = "",
    val trainer: TrainerModel = TrainerModel(),
    val videos: List<VideoModel> = emptyList()
): Parcelable
```
that will be sent to `VideosFragment`

Comment (VivekV95) [reply app/src/main/java/com/trx/consumer/models/params/VideoParamsModel.kt:8]: @onodude Confirming if this is correct

File: app/src/main/res/layout/fragment_videos.xml [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+                    app:layout_constraintStart_toEndOf="@+id/imgTrainerProfile"
+                    app:layout_constraintTop_toTopOf="parent"
+                    tools:text="Jamie Meneses" />
+
+                <com.trx.consumer.common.CommonButton
+                    android:id="@+id/btnTrainerProfile"
+                    android:layout_width="70dp"
+                    android:layout_height="30dp"
+                    android:layout_marginEnd="15dp"
+                    android:backgroundTint="@color/white"
```
Comment (VivekV95) [inline app/src/main/res/layout/fragment_videos.xml:128]: Not needed as background it already white

File: app/src/main/res/layout/fragment_videos.xml [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+                    android:textColor="@color/black"
+                    android:textSize="16sp"
+                    app:layout_constraintBottom_toBottomOf="parent"
+                    app:layout_constraintEnd_toStartOf="@+id/btnTrainerProfile"
+                    app:layout_constraintStart_toEndOf="@+id/imgTrainerProfile"
+                    app:layout_constraintTop_toTopOf="parent"
+                    tools:text="Jamie Meneses" />
+
+                <com.trx.consumer.common.CommonButton
+                    android:id="@+id/btnTrainerProfile"
```
Comment (VivekV95) [inline app/src/main/res/layout/fragment_videos.xml:124]: Looks like this needs a border according to the [design](https://www.figma.com/file/wMfhxL9w3KOw6okQJtt1OD/All-Screens-(v1.0.0)?node-id=2630%3A2), can use `app:outline="@color/black"`

File: app/src/main/res/layout/fragment_videos.xml [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+
+            <com.trx.consumer.common.CommonView
+                android:id="@+id/viewHeader"
+                android:layout_width="match_parent"
+                android:layout_height="385dp"
+                app:layout_constraintTop_toTopOf="parent">
+
+                <com.trx.consumer.common.CommonImageView
+                    android:id="@+id/imgHeader"
+                    android:layout_width="match_parent"
```
Comment (VivekV95) [inline app/src/main/res/layout/fragment_videos.xml:26]: Would be good to give this a slight alpha to make sure the title is visible (as the title color is white and a lot of the videos/workouts seem to have white backgrounds in their poster). We can do the same thing we're doing in the row view: `android:tint="@color/blackAlpha20"`

File: app/src/main/res/layout/fragment_videos.xml [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+                tools:itemCount="5"
+                tools:listitem="@layout/row_video_detail" />
+
+
+        </com.trx.consumer.common.CommonView>
+
+
+    </androidx.core.widget.NestedScrollView>
+
+    <com.trx.consumer.common.CommonHudView
```
Comment (VivekV95) [inline app/src/main/res/layout/fragment_videos.xml:216]: Can remove this, don't think this screen will need a hud

File: app/src/main/java/com/trx/consumer/models/params/VideoParamsModel.kt (hunk 1 above)
Comment (VivekV95) [reply app/src/main/java/com/trx/consumer/models/params/VideoParamsModel.kt:8]: The first video in the list can be used for the main section, and the other videos can be used for the adapter

File: app/src/main/java/com/trx/consumer/screens/videos/VideosFragment.kt [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+    private val handleLoadView = Observer<WorkoutModel> {
+        viewBinding.apply {
+            imgHeader.load(it.imageUrl)
+            lblTitle.text = it.video.name
+            lblSubtitle.text = it.duration
+            lblRelatedItems.text = "Related Workouts"
+            btnPrimary.text = "Start Workout"
+            lblTrainerName.text = it.trainer.fullName
+            lblSummary.text = it.trainer.bio
```
Comment (VivekV95) [inline app/src/main/java/com/trx/consumer/screens/videos/VideosFragment.kt:60]: `VideoModel` should have a property `description` that we can add to it and use here

Comment (VivekV95) [reply app/src/main/java/com/trx/consumer/screens/videos/VideosFragment.kt:60]: Nevermind on the above, we can handle this in a consolidation task

File: app/src/main/java/com/trx/consumer/models/params/VideoParamsModel.kt (hunk 1 above)
Comment (VivekV95) [inline app/src/main/java/com/trx/consumer/models/params/VideoParamsModel.kt:8]: Don't need the extra params model, can just send `WorkoutModel` directly

**Review by VivekV95 — CHANGES_REQUESTED** [review]
Left a few comments

File: app/src/main/res/layout/fragment_videos.xml (hunk 2 above)
Comment (halilyesilyurt) [reply app/src/main/res/layout/fragment_videos.xml]: Completed.

File: app/src/main/res/layout/fragment_videos.xml (hunk 4 above)
Comment (halilyesilyurt) [reply app/src/main/res/layout/fragment_videos.xml:26]: Completed.

File: app/src/main/res/layout/fragment_videos.xml (hunk 3 above)
Comment (halilyesilyurt) [reply app/src/main/res/layout/fragment_videos.xml:125]: Completed.

File: app/src/main/res/layout/fragment_videos.xml (hunk 5 above)
Comment (halilyesilyurt) [reply app/src/main/res/layout/fragment_videos.xml]: Completed.

File: app/src/main/java/com/trx/consumer/models/params/VideoParamsModel.kt (hunk 1 above)
Comment (halilyesilyurt) [reply app/src/main/java/com/trx/consumer/models/params/VideoParamsModel.kt]: i removed VideoParamsModel and use WorkoutModel directly as you said.

**Comment by halilyesilyurt** [comment]
changed pr images and demo.

**Comment by halilyesilyurt** [comment]
changed base branch to develop.

**Review by VivekV95 — DISMISSED** [review]

**Review by onodude — APPROVED** [review]

**Pull request merged**